### PR TITLE
[master] Allow skipping docker installation

### PR DESCRIPTION
**Pull request by PennyScissors**
Issue: https://github.com/rancher/rancher/issues/27646

~~As discussed, removed the docker installation step during provision. The expectation now is the user has already pre-installed everything they need to have installed docker themselves (will need to be documented).~~

Edit:
Updated provisioning to allow users to opt-in for skipping docker install by passing in `"none"` as the `engineInstallURL`, which can be easily added to the UI's dropdown of docker installation options. 
Users will now have the following options:
1. Provide no URL => Defaults to installing docker using default URL (get.docker.com)
2. Provide a URL => URL is used to install docker
3. Provide `"none"` as the URL => installing docker will be skipped

### CODE REVIEW

**Review by superseb — CHANGES_REQUESTED** [review]
I think there is a interpretation issue on the issue comment, the solution is to have a possibility to skip: ```basics here should be a skip provision for the create command which will bypass all provisioning and docker installation. ```

If we default to not provisioning, we will break all new node driver machines.

**Comment by vincent99** [comment]
I agree, this seems like it should be an opt-in option to the provisioner, or a different provisioner altogether (this is rancheros nodes skip installing Docker).

Almost all usage of almost all drivers produces an machine that still needs Docker installed on it (at least until doing rke2 provisioning).

If nothing else, we still have to maintain this for 2.5.x/2.4/2.3 for a long time.. the behavior certainly can't majorly change for those, so just dropping it would require we start maintaining two branches.

**Review by superseb — COMMENTED** [review]
I assume this is tested via the UI? So there is no input verification for an URL in this field?

**Comment by PennyScissors** [comment]
@superseb Yes, tested in the UI, all scenarios I listed work as expected.

**Review by superseb — DISMISSED** [review]

File: libmachine/provision/debian.go [hunk 1]
```diff
@@ -110,9 +111,13 @@ func (provisioner *DebianProvisioner) Provision(swarmOptions swarm.Options, auth
 		}
 	}
 
-	log.Debug("installing docker")
-	if err := installDockerGeneric(provisioner, engineOptions.InstallURL); err != nil {
-		return err
+	if strings.EqualFold(engineOptions.InstallURL, "none") {
```
Comment (luthermonson) [inline libmachine/provision/debian.go:114]: removing the yum update you can put this in installDockerGeneric and have one check

File: libmachine/provision/utils.go [hunk 2]
```diff
@@ -25,10 +25,15 @@ type DockerOptions struct {
 }
 
 func installDockerGeneric(p Provisioner, baseURL string) error {
+	if strings.EqualFold(baseURL, "none") {
+		log.Info("Skipping Docker installation")
+		return nil
+	}
 	// install docker - until cloudinit we use ubuntu everywhere so we
 	// just install it using the docker repos
+	log.Info("Installing Docker")
```
Comment (luthermonson) [inline libmachine/provision/utils.go:34]: maybe nice to put the baseURL in the log msg?

**Review by luthermonson — APPROVED** [review]

**Pull request merged**